### PR TITLE
Add get-extension-context to joyride.core

### LINKED ISSUE #33: Give the scripts access to the Joyride extension context
When extending VS Code you often need the [Extension Context](https://code.visualstudio.com/api/references/vscode-api#commands) instance. Joyride scripts needs this to take fuller control of VS Code.

This is somewhat related to:

* #8

### PR DESCRIPTION
**Pull request by PEZ**
Adds a function to `joyride.core` for getting the Joyride extension context.

Fixes #33

### CODE REVIEW

**Comment by borkdude** [comment]
I'll have a look tomorrow!

File: src/main/joyride/sci.cljs [hunk 1]
```diff
@@ -20,7 +22,9 @@
            (sci/init {:classes {'js goog/global
                                 :allow :all}
                       :namespaces (assoc (:namespaces pconfig/config)
-                                         'joyride.core {'*file* sci/file})
+                                         'joyride.core {'*file* sci/file
+                                                        'get-extension-context (fn []
```
Comment (borkdude) [inline src/main/joyride/sci.cljs:26]: PR looks good, but I'd prefer if get-extension context was a "true" var inside the SCI config. You can create it by defining `get-extension-context` with `defn` and then use `(sci/copy-var get-extension (sci/create-ns 'joyride.core))`

Comment (borkdude) [reply src/main/joyride/sci.cljs:26]: This would also allow you to add a docstring to it.

Comment (borkdude) [reply src/main/joyride/sci.cljs:26]: And get it using `clojure.repl/doc`.

Comment (PEZ) [reply src/main/joyride/sci.cljs:26]: Thanks! Where would I do the copying?

Comment (borkdude) [reply src/main/joyride/sci.cljs:26]: Look around in nbb.core for copy-var and you'll find an example. Sorry, tight schedule today!

Comment (PEZ) [reply src/main/joyride/sci.cljs]: I had already started poking around there. I think I have fixed it in a proper way now. And doc lookup works as well, added it to the Joyride API example while at it.

Good to merge?

**Comment by PEZ** [comment]
I took the ”PR looks good” hint. 😄

**Pull request merged**